### PR TITLE
restore cacher event Gone tests

### PR DESCRIPTION
**Pull request by gongguan**
Ref #90058

/kind cleanup

```release-note
None
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go [hunk 1]
```diff
@@ -100,6 +106,10 @@ func newEtcdTestStorage(t *testing.T, prefix string) (*etcd3testing.EtcdTestServ
 }
 
 func newTestCacher(s storage.Interface, cap int) (*cacherstorage.Cacher, storage.Versioner, error) {
+	return buildCacherWithClock(s, cap, clock.RealClock{})
+}
+
+func buildCacherWithClock(s storage.Interface, cap int, clock clock.Clock) (*cacherstorage.Cacher, storage.Versioner, error) {
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go:112]: nit: let's name it newTestCacherWithClock

File: staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go [hunk 2]
```diff
@@ -112,11 +122,18 @@ func newTestCacher(s storage.Interface, cap int) (*cacherstorage.Cacher, storage
 		NewFunc:        func() runtime.Object { return &example.Pod{} },
 		NewListFunc:    func() runtime.Object { return &example.PodList{} },
 		Codec:          codecs.LegacyCodec(examplev1.SchemeGroupVersion),
+		Clock:          clock,
 	}
 	cacher, err := cacherstorage.NewCacherFromConfig(config)
 	return cacher, v, err
 }
 
+func newTestCacherWithFakeClock(s storage.Interface, cap int) (*cacherstorage.Cacher, storage.Versioner, *clock.FakeClock, error) {
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go:131]: Let's remove this function, it's cleaner if you create the clock yourself in the single test where you need it and simply pass it here.

**Review by wojtek-t — COMMENTED** [review]
Two small comments - other than that lgtm

File: staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go (hunk 1 above)
Comment (gongguan) [reply staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go]: done.

File: staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go (hunk 2 above)
Comment (gongguan) [reply staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go]: done, it's cleaner.

**Comment by wojtek-t** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**